mne/viz/topomap.py

```diff
@@ -1578,11 +1572,267 @@ def _prepare_topomap(pos, ax):
+                   aspect='equal', extent=(xmin, xmax, ymin, ymax),
+                   interpolation='bilinear')
+    plt.colorbar(im, cax=ax_cbar, cmap=cmap)
+    cont = ax.contour(Xi, Yi, Zi, levels=cont_lims, colors='k', linewidths=1)
+
+    patch_ = patches.Ellipse((0, 0),
+                             2 * outlines['clip_radius'][0],
+                             2 * outlines['clip_radius'][1],
+                             clip_on=True,
+                             transform=ax.transData)
+    im.set_clip_path(patch_)
+    text = ax.text(0.55, 0.95, '', transform=ax.transAxes, va='center',
+                   ha='right')
+    params['text'] = text
+    items.append(im)
+    items.append(text)
+    for col in cont.collections:
+        col.set_clip_path(patch_)
+
+    outlines_ = dict([(k, v) for k, v in outlines.items() if k not in
+                      ['patch', 'autoshrink']])
+    for k, (x, y) in outlines_.items():
+        if 'mask' in k:
+            continue
+        ax.plot(x, y, color='k', linewidth=1, clip_on=False)
+    params.update({'patch': patch_, 'outlines': outlines_})
+    return tuple(items) + tuple(cont.collections)
+
+
+def _animate(frame, ax, ax_line, params):
+    """Updates animated topomap."""
+    if params['pause']:
+        frame = params['frame']
+    time_idx = params['frames'][frame]
+
+    title = '%6.0f ms' % (params['times'][frame] * 1e3)
+    if params['blit']:
+        text = params['text']
+    else:
+        ax.cla()  # Clear old contours.
+        text = ax.text(0.45, 1.15, '', transform=ax.transAxes)
+        for k, (x, y) in params['outlines'].items():
+            if 'mask' in k:
+                continue
+            ax.plot(x, y, color='k', linewidth=1, clip_on=False)
+
+    _hide_frame(ax)
+    text.set_text(title)
+
+    vmin = params['vmin']
+    vmax = params['vmax']
+    Xi = params['Xi']
+    Yi = params['Yi']
+    Zi = params['Zis'][frame]
+    extent = params['extent']
+    cmap = params['cmap']
+    patch = params['patch']
+
+    im = ax.imshow(Zi, cmap=cmap, vmin=vmin, vmax=vmax, origin='lower',
+                   aspect='equal', extent=extent, interpolation='bilinear')
+    cont_lims = params['cont_lims']
+    cont = ax.contour(Xi, Yi, Zi, levels=cont_lims, colors='k', linewidths=1)
+
+    im.set_clip_path(patch)
+    items = [im, text]
+    for col in cont.collections:
+        col.set_clip_path(patch)
+
+    if params['butterfly']:
+        all_times = params['all_times']
+        line = params['line']
+        line.remove()
+        params['line'], = ax_line.plot([all_times[time_idx],
+                                        all_times[time_idx]],
+                                       ax_line.get_ylim(), color='r')
+        items.append(params['line'])
+    params['frame'] = frame
+    return tuple(items) + tuple(cont.collections)
+
+
+def _pause_anim(event, params):
+    """Function for pausing and continuing the animation on mouse click"""
+    params['pause'] ^= True
```

Comment:
so it's the xor operator? Does `params['pause']` take only `True` and `False` values? I gave it a try and it appears to be equivalent to the complement operator at least in this scenario. Maybe we should use the simpler (and more familiar) operator?